### DISCUSSION TITLE
Suspend/resume autohide timeout in a good way

### DISCUSSION THREAD

**Discussion by przepompownia** [folke/noice.nvim#281] [Q&A] [answered]
Currently I have 
```lua
  views = {
    mini = {
      timeout = 5000,
    },
  },
```
but for some messages (e.g. the output from `:Inspect`) it would be good to pause their timer used to autohide and resume it again with some other keymap. 

I accidentally found that switching to commandline allows to suspend the last message. 
![image](https://user-images.githubusercontent.com/11404453/209016530-948cfd3d-f653-43c8-9ad7-8f6b03069315.png)

Is it possible without such tricks?

**Comment by folke** [comment, marked as answer]
Check https://github.com/folke/noice.nvim#%EF%B8%8F-command-redirection

I use that cmdline keymap

**Reply by przepompownia** [reply]
It seems to be the solution for the output of commands, thanks @folke!

On the other side, the problem still exists for messages from vim.notify etc. 

On the third side I tested your tip on TUI because nvim-qt crashed repeatedly so I need to prepare a bug report there.

**Reply by przepompownia** [reply]
Testing the keymap on tmux inside kitty with 
```lua
vim.keymap.set('c', '<A-CR>', function()
  require('noice').redirect(vim.fn.getcmdline())
end, { desc = 'Redirect Cmdline' })
```
(Shift-Enter did not work there) I noticed that the input widget does not disappear after submitting with this shortcut. 
![image](https://user-images.githubusercontent.com/11404453/209019724-61295b8d-be24-4d34-9570-04de65d678d8.png)


To use clear output `lua require("noice").redirect("Inspect")` seems to be the best way at the moment.

**Reply by folke** [reply]
The input widget remaining is a feature. It's useful to run the same command again and again. If you don't want that behavior, you could also add something like `vim.api.nvim_input("<esc>")` to that keybinding

**Reply by przepompownia** [reply]
It seems to resolve the cmdline output part!

**Comment by przepompownia** [comment]
Crash was reported in https://github.com/equalsraf/neovim-qt/issues/1043
